apptrust/commands/version/create_app_version_cmd_test.go

```diff
@@ -928,3 +928,561 @@ func TestValidateRequiredFieldsInMap(t *testing.T) {
+					},
+				},
+				Excluded: []*model.CreateVersionSourceFilter{
+					{
+						Path: "libs/vulnerable-*.jar",
+					},
+				},
+			},
+		},
+		{
+			name: "filters from flags when spec flag is not set",
+			ctxSetup: func(ctx *components.Context) {
+				ctx.Arguments = []string{"app-key", "1.0.0"}
+				ctx.AddStringFlag(commands.SourceTypePackagesFlag, "type=npm,name=test,version=1.0.0,repo-key=repo")
+				ctx.AddStringFlag(commands.IncludeFilterFlag, "filter_type=package, type=docker, name=frontend-*")
+				ctx.AddStringFlag("url", "https://example.com")
+			},
+			expectedFilters: &model.CreateVersionFilters{
+				Included: []*model.CreateVersionSourceFilter{
+					{
+						PackageType: "docker",
+						PackageName: "frontend-*",
+					},
+				},
+			},
+		},
+		{
+			name: "no filters when spec has no filters and no filter flags",
+			ctxSetup: func(ctx *components.Context) {
+				ctx.Arguments = []string{"app-key", "1.0.0"}
+				ctx.AddStringFlag(commands.SpecFlag, "./testfiles/minimal-spec.json")
+				ctx.AddStringFlag("url", "https://example.com")
+			},
+			expectedFilters: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := &components.Context{}
+			tt.ctxSetup(ctx)
+
+			payload, err := cmd.buildRequestPayload(ctx)
+			if tt.expectError {
+				assert.Error(t, err)
+				if tt.errorContains != "" {
+					assert.Contains(t, err.Error(), tt.errorContains)
+				}
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, payload)
+				if tt.expectedFilters == nil {
+					assert.Nil(t, payload.Filters)
+				} else {
+					assert.NotNil(t, payload.Filters)
+					if tt.expectedFilters.Included != nil {
+						assert.Equal(t, len(tt.expectedFilters.Included), len(payload.Filters.Included))
+						for i, expected := range tt.expectedFilters.Included {
+							assert.Equal(t, expected.PackageType, payload.Filters.Included[i].PackageType)
+							assert.Equal(t, expected.PackageName, payload.Filters.Included[i].PackageName)
+							assert.Equal(t, expected.PackageVersion, payload.Filters.Included[i].PackageVersion)
+							assert.Equal(t, expected.Path, payload.Filters.Included[i].Path)
+							assert.Equal(t, expected.SHA256, payload.Filters.Included[i].SHA256)
+						}
+					}
+					if tt.expectedFilters.Excluded != nil {
+						assert.Equal(t, len(tt.expectedFilters.Excluded), len(payload.Filters.Excluded))
+						for i, expected := range tt.expectedFilters.Excluded {
+							assert.Equal(t, expected.PackageType, payload.Filters.Excluded[i].PackageType)
+							assert.Equal(t, expected.PackageName, payload.Filters.Excluded[i].PackageName)
+							assert.Equal(t, expected.PackageVersion, payload.Filters.Excluded[i].PackageVersion)
+							assert.Equal(t, expected.Path, payload.Filters.Excluded[i].Path)
+							assert.Equal(t, expected.SHA256, payload.Filters.Excluded[i].SHA256)
+						}
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestValidateNoSpecAndFlagsTogether_WithFilterFlags(t *testing.T) {
```

Comment:
Just for having it ordered, I suggest moving this test to be below the `TestValidateNoSpecAndFlagsTogether` above